airbyte-integrations/connectors/source-bing-ads/source_bing_ads/source.py

```diff
@@ -0,0 +1,302 @@
+    AdGroup schema: https://docs.microsoft.com/en-us/advertising/campaign-management-service/adgroup?view=bingads-13
+    """
+
+    data_field: str = "AdGroup"
+    service_name: str = "CampaignManagement"
+    operation_name: str = "GetAdGroupsByCampaignId"
+    additional_fields: str = "AdGroupType AdScheduleUseSearcherTimeZone CpmBid CpvBid MultimediaAdsBidAdjustment"
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        return {"CampaignId": stream_slice["campaign_id"], "ReturnAdditionalFields": self.additional_fields}
+
+    def stream_slices(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        campaigns = Campaigns(self.client, self.config)
+        for account_id in self.config["account_ids"]:
+            for campaign in campaigns.read_records(sync_mode=SyncMode.full_refresh, stream_slice={"account_id": account_id}):
+                yield {"campaign_id": campaign["Id"], "account_id": account_id}
+
+        yield from []
+
+
+class Ads(BingAdsStream):
+    """
+    Retrieves the ads for all provided accounts.
+    API doc: https://docs.microsoft.com/en-us/advertising/campaign-management-service/getadsbyadgroupid?view=bingads-13
+    Ad schema: https://docs.microsoft.com/en-us/advertising/campaign-management-service/ad?view=bingads-13
+    """
+
+    data_field: str = "Ad"
+    service_name: str = "CampaignManagement"
+    operation_name: str = "GetAdsByAdGroupId"
+    additional_fields: str = "ImpressionTrackingUrls Videos LongHeadlines"
+    ad_types = ["Text", "Image", "Product", "AppInstall", "ExpandedText", "DynamicSearch", "ResponsiveAd", "ResponsiveSearch"]
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        return {
+            "AdGroupId": stream_slice["ad_group_id"],
+            "AdTypes": {"AdType": self.ad_types},
+            "ReturnAdditionalFields": self.additional_fields,
+        }
+
+    def stream_slices(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        ad_groups = AdGroups(self.client, self.config)
+        for slice in ad_groups.stream_slices(sync_mode=SyncMode.full_refresh):
+            for ad_group in ad_groups.read_records(sync_mode=SyncMode.full_refresh, stream_slice=slice):
+                yield {"ad_group_id": ad_group["Id"], "account_id": slice["account_id"]}
+        yield from []
+
+
+class SourceBingAds(AbstractSource):
```

Comment:
Could you add a docstring comment to a source with what BingAds does. It helps understand what happened in code on a high level.

For example: """Source Intercom fetch data from messaging platform."""